pkg/api/og.go

```diff
@@ -0,0 +1,462 @@
+				}
+			}
+		}
+
+		// Apply circular mask using DrawMask
+		maskedAvatar := image.NewRGBA(image.Rect(0, 0, avatarSize, avatarSize))
+		imagedraw.DrawMask(maskedAvatar, maskedAvatar.Bounds(), scaledAvatar, image.Point{}, mask, image.Point{}, imagedraw.Over)
+
+		// Add circular border
+		avatarCenterX := imageX + avatarDisplaySize/2
+		avatarCenterY := imageY + avatarDisplaySize/2
+		canvasCtx.SetStrokeColor(imageBorderColor)
+		canvasCtx.SetStrokeWidth(3)
+		canvasCtx.DrawPath(avatarCenterX, avatarCenterY, canvas.Circle(avatarDisplaySize/2))
+		canvasCtx.Stroke()
+
+		// Draw the final circular avatar
+		canvasCtx.DrawImage(imageX, imageY, maskedAvatar, canvas.DPMM(canvasDPMM))
+	}
+
+	// Create unified responsive "Join @handle" text
+	joinUserContent := fmt.Sprintf("Join @%s", handle)
+
+	availableWidth := textAvailableWidth // Full available width for the text
+	joinText, _ := createResponsiveJoinText(fontAHN, joinUserContent, availableWidth)
+	canvasCtx.DrawText(textStartX, joinY, joinText)
+
+	// Add "streaming on Stream.place" subtitle
+	onFace := fontAHN.Face(subtitleFontSize, subtitleColor, canvas.FontRegular, canvas.FontNormal)
+	onText := canvas.NewTextBox(onFace, "streaming on Stream.place", 250, 30, canvas.Left, canvas.Center, &canvas.TextOptions{})
+	canvasCtx.DrawText(textStartX, subtitleY, onText)
+
+	// Add user description or promotional text
+	descFace := fontAHN.Face(descFontSize, descColor, canvas.FontRegular, canvas.FontNormal)
+	descText := canvas.NewTextBox(descFace, description, 230, 30, canvas.Left, canvas.Center, &canvas.TextOptions{})
+	canvasCtx.DrawText(textStartX, descY, descText)
+
+	b := &bytes.Buffer{}
+	if err := c.Write(b, renderers.JPEG(canvas.DPMM(canvasDPMM))); err != nil {
+		return nil, fmt.Errorf("failed to render canvas to buffer: %w", err)
+	}
+
+	return b.Bytes(), nil
+}
+
+func (a *StreamplaceAPI) fetchUserProfile(ctx context.Context, username string) (*bsky.ActorDefs_ProfileViewDetailed, error) {
+	// Use ATSync to resolve username to DID, then fetch full profile from Bluesky
+	var actor string
+
+	// First try to resolve via internal DB
+	repo, err := a.ATSync.Model.GetRepoByHandleOrDID(username)
+	if err != nil {
+		// log.Warn(ctx, "failed to resolve via ATSync, trying username directly", "username", username, "error", err)
+		// // Fall back to using the username directly
+		// actor = username
+		// if !strings.HasPrefix(username, "did:") && !strings.Contains(username, ".") {
+		// 	actor = username + ".bsky.social"
+		// }
+
+		return nil, fmt.Errorf("%w: %w", ErrUserNotFound, err)
+	} else if repo != nil {
+		// Use the DID as it's the most reliable identifier
+		actor = repo.DID
+	} else {
+		return nil, fmt.Errorf("no repo found for username: %s (%w)", username, ErrUserNotFound)
+	}
+
+	// TODO: check if actor is restricted
+
+	// Fetch full profile from Bluesky public API
+	client := &xrpc.Client{
+		Host: "https://public.api.bsky.app",
+	}
```

Comment:
Fine for now but please add a reference to this in https://github.com/streamplace/streamplace/issues/195 so we know to fix this usage too